### DISCUSSION TITLE
Weird behavior when wrongly combining `Annotated` with `List`

### DISCUSSION THREAD

**Discussion by The-Compiler** [fastapi/typer#713] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing_extensions import Annotated
from typing import List, get_type_hints

import typer


def test(arg: List[Annotated[str, typer.Argument(help="test")]]):  # wrong!
    print("test")


if __name__ == "__main__":
    print(get_type_hints(test))
    typer.run(test)
```

### Description

On Python 3.8, this results in:

```
AssertionError: List types with complex sub-types are not currently supported
```

from here:

https://github.com/tiangolo/typer/blob/4c6098f9fe893cbe5536b1cb793ef7dd35bfd7c1/typer/main.py#L834-L836

On newer Python versions, this works, but the `typer.Argument` silently gets dropped. See `--help`:

```
*    arg      ARG...  [default: None] [required]
```

as opposed to using `Annotated[List[str], typer.Argument(help="test")]` correctly:

```
*    arg      ARG...  test [default: None] [required]  
```

---

The fact that Python 3.9+ fails silently is due to [`typing.get_type_hints()`](https://docs.python.org/3/library/typing.html#typing.get_type_hints) by default [stripping metadata](https://peps.python.org/pep-0593/#interaction-with-get-type-hints) there. Thus, when it's called here:

https://github.com/tiangolo/typer/blob/4c6098f9fe893cbe5536b1cb793ef7dd35bfd7c1/typer/utils.py#L110

the `Annotated` part gets dropped here:

https://github.com/tiangolo/typer/blob/4c6098f9fe893cbe5536b1cb793ef7dd35bfd7c1/typer/utils.py#L164-L166

### Operating System

Linux

### Typer Version

0.9.0

### Python Version

3.8.18, 3.9.18, 3.11.6

**Comment by lachaib** [comment, marked as answer]
Actually, I would say the usage is wrong here.
It makes no sense to annotate the item type of the list with the `typer.Argument` description.

Your code would work just fine if you had done the reverse:
```python
def test(arg: Annotated[List[str], typer.Argument(help="test")]):  #  <- yay!
    print("test")
 ```

**Reply by The-Compiler** [reply]
I agree, hence the `# wrong!` comment in the code. But it silently being ignored can be a very confused outcome to spot and/or debug.
